msal/src/main/java/com/microsoft/identity/client/SingleAccountPublicClientApplication.java

```diff
@@ -66,10 +107,100 @@ private void initializeSharedPreferenceFileManager(@NonNull final Context contex
                 new StorageHelper(context));
     }
 
+
     @Override
-    public void getCurrentAccount(@NonNull final CurrentAccountCallback callback) {
+    public void getCurrentAccount(final CurrentAccountCallback callback) {
+        final String methodName = ":getCurrentAccount";
         final PublicClientApplicationConfiguration configuration = getConfiguration();
 
+
+        try {
+            if (MSALControllerFactory.brokerEligible(
```

Comment:
brokerEligible() is already in BrokerMsalController.getCurrentAccount

Reply:
(We might also want to rename this function and its .aidl counterpart to avoid confusion)

Reply:
Here's i'm opting between shared mode and non shared mode.  I suppose I can assume that if the device is in shared mode that the broker is installed.

Reply:
Fixed.